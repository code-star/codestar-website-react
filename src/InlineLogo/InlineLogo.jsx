import React, { Component } from 'react';

class TextLogo extends Component {
  render() {
    const { dark, small } = this.props;
    return (
      <img
        src={`${process.env.PUBLIC_URL}/images/codestar_logo_${
          dark ? 'dark' : 'light'
        }.svg`}
        alt="Codestar"
        style={
          small
            ? { height: '0.8em', marginBottom: '0.2em' }
            : { height: '0.8em', marginBottom: '0.12em' }
        }
      />
    );
  }
}

class InlineLogo extends Component {
  replaceWithLogo(text) {
    return text
      .split('Codestar')
      .reduce(
        (accu, t, i) =>
          i === 0
            ? [t]
            : accu.concat(
                <TextLogo dark={this.props.dark} small={this.props.small} />,
                t
              ),
        []
      );
  }

  injectLogo(children) {
    return React.Children.map(children, (child, i) => {
      if (!child.props) {
        return this.replaceWithLogo(child);
      } else if (child.props.children) {
        return React.cloneElement(child, {
          children: this.injectLogo(child.props.children),
        });
      } else {
        return React.cloneElement(child);
      }
    });
  }

  render() {
    return this.injectLogo(this.props.children);
  }
}

export default InlineLogo;
export { TextLogo };
