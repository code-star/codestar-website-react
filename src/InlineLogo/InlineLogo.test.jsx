import React from 'react';
import ReactDOM from 'react-dom';
import InlineLogo from './InlineLogo';
import renderer from 'react-test-renderer';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <InlineLogo>
      <div>test</div>
    </InlineLogo>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('replaces Codestar in a string with an image', () => {
  const comp = renderer
    .create(
      <InlineLogo>
        <div>test Codestar test</div>
      </InlineLogo>
    )
    .toJSON();
  expect(comp).toMatchSnapshot();
});

it('leaves string without Codestar unchanged', () => {
  const comp = renderer
    .create(
      <InlineLogo>
        <div>test test</div>
      </InlineLogo>
    )
    .toJSON();
  expect(comp).toMatchSnapshot();
});
