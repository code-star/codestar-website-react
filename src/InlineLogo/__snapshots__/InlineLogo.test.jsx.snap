// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`leaves string without Codestar unchanged 1`] = `
<div>
  test test
</div>
`;

exports[`replaces Codestar in a string with an image 1`] = `
<div>
  test 
  <img
    alt="Codestar"
    src="/images/codestar_logo_light.svg"
    style={
      Object {
        "height": "0.8em",
        "marginBottom": "0.12em",
      }
    }
  />
   test
</div>
`;
